src/ui/components/Rating/index.js

```diff
@@ -68,6 +68,7 @@ export class RatingBase extends React.Component {
       };
 
       if (readOnly) {
+        delete props.id;
```

Comment:
I'm less a fan of deleting the prop and rather only adding the ID to the object if `readOnly` is false.

Also, you'll need to add a test for this so you can make sure it's working as intended 😄